test/blackbox-tests/test-cases/pkg/portable-lockdirs/portable-lockdirs-custom-platforms.t

```diff
@@ -0,0 +1,140 @@
+Specifying custom platforms to solve for.
+
+  $ . ../helpers.sh
+  $ mkrepo
+  $ add_mock_repo_if_needed
+
+Create a package that writes a different value to some files depending on the os, with a non-defaut os.
+  $ mkpkg foo <<EOF
+  > build: [
+  >   ["mkdir" "-p" share "%{lib}%/%{name}%"]
+  >   ["touch" "%{lib}%/%{name}%/META"] # needed for dune to recognize this as a library
+  >   ["sh" "-c" "echo OpenBSD > %{share}%/kernel"] { os = "openbsd" }
+  >   ["sh" "-c" "echo Other > %{share}%/kernel"] { os != "openbsd" }
+  > ]
+  > EOF
+
+Create a custom dune-workspace to solve for openbsd.
+  $ cat > dune-workspace <<EOF
+  > (lang dune 3.18)
+  > (repository
+  >  (name mock)
+  >  (url "file://$(pwd)/mock-opam-repository"))
+  > (lock_dir
+  >  (repositories mock)
+  >  (solve_for_platforms
+  >   (extra
```

Comment:
This "extra" field isn't really consistent with the rest of the dune language. I suggest that you use a field like `platforms` and use the OSL to add new configurations.

If this is too much work, you can always just allow for overriding the entire field for now and revisit OSL support later.

Reply:
Ok I'll change it so specifying a custom list of platforms always overrides the default list of platforms. I initially tried to use OSL for this but I couldn't figure out how to use OSL when the elements of the set need to be parsed with `Dune_lang.Decoder`.